packages/lit-dev-content/site/articles/article/redux-reactive-controllers.md

```diff
@@ -0,0 +1,731 @@
+
+Next we need to give some elements the ability to connect and subscribe to changes to the store and update the UI. On top of that we will write a “Selector” which will select a specific datum from the overall state, and if that value changes, we will tell the Reactive Controller host to re-render.
+
+### Attaching the Controller and Accepting options
+
+First we will write the definition of the `SelectorController` class and attach it to the host element so that the Reactive Controller can hook into the host’s reactive update lifecycle:
+
+```ts
+import type { ReactiveController, ReactiveControllerHost } from 'lit';
+
+export class SelectorController implements ReactiveController {
+  host: ReactiveControllerHost;
+
+  constructor(host: ReactiveControllerHost) {
+    this.host = host;
+    host.addController(this);
+  }
+}
+```
+
+Next, let's accept the following options: the Redux store in which we would like to subscribe to, as well as the Selector we would like to use to select what data we would like to render from the store:
+
+```ts
+import type { Store, Action as RAction } from '@reduxjs/toolkit';
+...
+
+export type Selector<State, Result> = (state: State) => Result;
+
+export class SelectorController<
+    State,
+    Action extends RAction,
+    Result = unknown
+> implements ReactiveController {
+  host: ReactiveControllerHost;
+  store: Store<State, Action>;
+  selector: Selector<State, Result>;
+
+  constructor(
+    host: ReactiveControllerHost,
+    store: Store<State, Action>,
+    selector: Selector<State, Result>,
+  ) {
+    this.host = host;
+    host.addController(this);
+
+    this.store = store;
+    this.selector = selector;
+  }
+}
+```
+
+Now let’s initialize the initially selected value so that the user can access the state’s initial value with `selectorController.selected` using the user's selector:
+
+```ts
+...
+export class SelectorController<
+    State,
+    Action extends RAction,
+    Result = unknown
+> implements ReactiveController {
+  ...
+  selected: Result;
```

Comment:
Nit: I think `value` makes more sense for this field which also should be marked public.